libraries/chain/wasm_interface.cpp

```diff
@@ -1888,6 +1916,12 @@ REGISTER_INTRINSICS(bandwith_api,
     (confirm_bw_limits, void(int64_t))
 );
 
+REGISTER_INTRINSICS(ram_provide_api,
+    (get_current_ram_provider, int64_t(int64_t))
+    (get_available_providers_count, int(int64_t))
+    (get_available_ram_provider, int64_t(int64_t, int))
+    (set_current_ram_provider, void(int64_t, int64_t))
```

Comment:
It will be too difficult to use this functionality in a smart contract. The best solution would be to provide one function: `get_ram_provider(account)`, which the provider will return if there is one, or an account if there is no provider for this account.